linode/manager
# feat: [M3-7841] - Add the capability to search for a Linode by ID using the main search tool.

## Description 📝
This PR add's capability to the main search tool to search linode by ID, example id: 1234 should show the matching result. 
Also fixes the issue of rendering correct icon for the buckets in the results list. 

## Target release date 🗓️
10/28

## Preview 📷

| Before  | After   |
| ------- | ------- |
|![image](https://github.com/user-attachments/assets/c03b43da-83c7-47d7-b338-52529cde1ae4)|![image](https://github.com/user-attachments/assets/b39c2457-32f2-4c71-b836-c6e11de64dbc)|
|![image](https://github.com/user-attachments/assets/d7c06b80-2772-4648-8251-54e05b4a00fc)|![image](https://github.com/user-attachments/assets/5aa28551-2571-416c-96dd-adb7ce47d439)|

## How to test 🧪

### Verification steps
(How to verify changes)
- Checkout and run locally 
- Use main search tool to search any linode by typing its id example (1234 or id:1234)
- Verfrify and confirm on bucket icon for storage bucket search result.
- Verify the results and no regression in main search functionality .

## As an Author I have considered 🤔

*Check all that apply*
